project/Build.scala

```diff
@@ -35,6 +35,8 @@ object BuildSettings {
     licenses      := ("Apache2", new java.net.URL("http://www.apache.org/licenses/LICENSE-2.0.txt")) :: Nil,
     homepage      := Some(new java.net.URL("http://www.softwaremill.com"))
   )
+
+  val scalacSettings = Seq("-deprecation","-feature","-unchecked","-Ybackend:GenBCode")
```

Comment:
I have it in all my code, because I was encountering some issues without it (I think with some complex pattern matching, or something like that. I guess it makes no sense in this project). Removed in next commit.